Sources/ApolloCodegenTestSupport/MockFileManager.swift

```diff
@@ -3,77 +3,113 @@ import ApolloCodegenLib
 import ApolloUtils
 import XCTest
 
-/// A `FileManagerProvider` that can be used to mock the `apollo` extension as  found on `FileManager`.
-public class MockFileManager: FileManagerProvider {
-  public typealias fileExistsBlock = ((String, UnsafeMutablePointer<ObjCBool>?) -> Bool)
-  public typealias removeItemBlock = ((String) throws -> ())
-  public typealias createFileBlock = ((String, Data?, FileAttributes?) -> Bool)
-  public typealias createDirectoryBlock = ((String, Bool, FileAttributes?) throws -> ())
-
-  public enum BlocksCalled {
-    case fileExists
-    case removeItem
-    case createFile
-    case createDirectory
+/// Used to mock a `FileManager` instance that is compatible with the `.apollo` namespace extension.
+public class MockFileManager: FileManager {
+  /// Translates to the `FileManager` functions that can be mocked.
+  public enum Closure: CustomStringConvertible {
+    case fileExists(_ handler: (String, UnsafeMutablePointer<ObjCBool>?) -> Bool)
+    case removeItem(_ handler: (String) throws -> Void)
+    case createFile(_ handler: (String, Data?, FileAttributes?) -> Bool)
+    case createDirectory(_ handler: (String, Bool, FileAttributes?) throws -> Void)
+
+    public var description: String {
```

Comment:
These descriptions match the output from the `#function` macro which is used in each overridden function to lookup the closure.

Reply:
Might be worth throwing a note in here about that in case for some stupid reason the `#function` macro changes in the future